batchflow/opensets/coco.py

```diff
@@ -0,0 +1,112 @@
+""" Datasets for COCO challenge tasks, http://cocodataset.org/#home.
+    Yet contains only the dataset for Semantic Segmentation. """
+
+import os
+import logging
+import tempfile
+from glob import glob
+from zipfile import ZipFile
+from os.path import dirname, basename
+
+import tqdm
+import requests
+from PIL import Image
+
+from . import ImagesOpenset
+from .. import FilesIndex, any_action_failed, parallel
+
+logger = logging.getLogger('COCO')
+
+
+class BaseCOCO(ImagesOpenset):
+    """ Base class for COCO datasets. """
+
+    TRAIN_IMAGES_URL = 'http://images.cocodataset.org/zips/train2017.zip'
+    TEST_IMAGES_URL = 'http://images.cocodataset.org/zips/val2017.zip'
+
+    def __init__(self, *args, preloaded=None, **kwargs):
+        super().__init__(*args, preloaded=preloaded, **kwargs)
+
+
+    @parallel(init='_get_from_urls', post='_post_fn', target='t')
+    def download(self, url, folder, train_val, path=None):
+        """ Download the archives and extracts it's content. Downloading is performed in parallel manner.
+        Set of URL's to download from is defined in the `_get_from_urls` method.
+        The aggregation of the content from all archives is performed in `_post_fn` method.
+        """
+        logger.info('Downloading %s', url)
+        if path is None:
+            path = tempfile.gettempdir()
+        filename = basename(url)
+        localname = os.path.join(path, filename)
+        if not os.path.isfile(localname):
+            r = requests.get(url, stream=True)
+            file_size = int(r.headers['Content-Length'])
+            chunk = 1
+            chunk_size = 1024 * 1000 #MBs
+            num_bars = int(file_size / chunk_size)
+            with open(localname, 'wb') as f:
+                for chunk in tqdm.tqdm(r.iter_content(chunk_size=chunk_size), total=num_bars,
+                                       unit='MB', desc=filename, leave=True):
+                    f.write(chunk)
+        return self._extract_if_not_exist(localname, folder, train_val)
+
+
+class COCOSegmentation(BaseCOCO):
```

Comment:
As far as I recall, it was you who added names for each of the classes in `Pascal` dataset, so why would not you do the same for `CoCo`?